### PR TITLE
Config log retention

### PR DESCRIPTION
**Pull request by noah-paige**
### Feature or Bugfix

- Feature

### Detail
- Configurable Log Retention via `cdk.json`

### Relates
- 

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/modules/mlstudio/cdk/mlstudio_extension.py [hunk 1]
```diff
@@ -63,7 +64,7 @@ def extent(setup: EnvironmentSetup):
                     setup,
                     f'SageMakerStudio{_environment.name}',
                     log_group_name=f'/{_environment.resourcePrefix}/{_environment.name}/vpc/sagemakerstudio',
-                    retention=logs.RetentionDays.ONE_MONTH,
+                    retention=getattr(logs.RetentionDays, os.environ.get('LOG_RETENTION', 'TWO_YEARS')),
```
Comment (dlpzx) [inline backend/dataall/modules/mlstudio/cdk/mlstudio_extension.py:67]: In the cdk.json we are expected users to input the log retention time as a string such as `ONE_MONTH`. but the parameter `retention` does not accept strings, it accepts logs.RetentionDays values which are values of an Enum: https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_logs/RetentionDays.html

File: deploy/stacks/pipeline.py [hunk 2]
```diff
@@ -43,6 +44,9 @@ def __init__(
         self.target_envs = target_envs
         self.repo_string = repo_string
         self.repo_connection_arn = repo_connection_arn
+        self.log_retention_duration = (
+            self.node.try_get_context('log_retention_duration') or logs.RetentionDays.TWO_YEARS.value
+        )
```
Comment (dlpzx) [inline deploy/stacks/pipeline.py:49]: Should we provide the enum item value as default or directly the enum key?

Comment (dlpzx) [reply deploy/stacks/pipeline.py:49]: Then we were doing it wrong in the mlstudio log group:

image

Comment (noah-paige) [reply deploy/stacks/pipeline.py:49]: They are handled the same:
```
>>> getattr(logs.RetentionDays, 'ONE_MONTH')
<RetentionDays.ONE_MONTH: 'ONE_MONTH'>
>>> logs.RetentionDays.ONE_MONTH
<RetentionDays.ONE_MONTH: 'ONE_MONTH'>
```

Comment (dlpzx) [reply deploy/stacks/pipeline.py:49]: Yeap, in this case it does not really affect

**Review by dlpzx — APPROVED** [review]
looks good

**Pull request merged**